onnx/defs/reduction/utils.cc

```diff
@@ -103,16 +100,14 @@ to `False` instead of `True`.)DOC";
         return;
       }
 
-      int64_t keep_dims = 1, noop_with_empty_axes = 0;
-      auto attr_proto = ctx.getAttribute("keepdims");
-      if (attr_proto) {
-        keep_dims = attr_proto->i();
-      }
-      auto noop_attr_proto = ctx.getAttribute("noop_with_empty_axes");
-      if (noop_attr_proto) {
-        noop_with_empty_axes = noop_attr_proto->i();
-      }
+      const auto& input_shape = ctx.getInputType(0)->tensor_type().shape();
+      const int64_t input_ndim = input_shape.dim_size();
+      auto output_shape = ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
+
+      const auto keep_dims = static_cast<int64_t>(ctx.getAttribute("keepdims")->i());
```

Comment:
what if keepdims is not specified? (the code you replaced was able to handle that).

Reply:
I suggest using `getAttribute(ctx, "keepdims", 1)`, which handles all that. Please see [here](https://github.com/onnx/onnx/blob/b2ee94b696a606514ed8c2a426cf562946a92ca2/onnx/defs/shape_inference.h#L153). Same for next line.